src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -639,31 +643,43 @@ void tryToConnect(NatsUri cur, NatsUri resolved, long now) {
                 this.currentServer = cur;
                 this.serverAuthErrors.clear(); // reset on successful connection
                 updateStatus(Status.CONNECTED); // will signal status change, we also signal in finally
-            } finally {
+            }
+            finally {
                 statusLock.unlock();
             }
             timeTraceLogger.trace("status updated");
-        } catch (Exception exp) {
+        }
+        catch (Exception exp) {
             processException(exp);
             try {
                 // allow force reconnect since this is pretty exceptional,
                 // a connection failure while trying to connect
                 this.closeSocket(false, true);
-            } catch (InterruptedException e) {
+            }
+            catch (InterruptedException e) {
                 processException(e);
                 Thread.currentThread().interrupt();
             }
-        } finally {
+        }
+        finally {
             statusLock.lock();
             try {
                 this.connecting = false;
                 statusChanged.signalAll();
-            } finally {
+            }
+            finally {
                 statusLock.unlock();
             }
         }
     }
 
+    private void clearCurrentServer() {
+        if (currentServer != null) {
+            lastServer = currentServer;
+        }

```

Comment:
I know this looks weird, but this makes sure lastServer is retained when clearServer is called when currentServer is already null, it can happen.